src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlNullValues.cs

```diff
@@ -0,0 +1,212 @@
+        private string UdfNameNotNull = DatabaseHelper.GenerateUniqueName("SqlNullValuesRetValNotNull");
+
+        public SqlNullValuesTests(SQLSetupStrategyCertStoreProvider fixture)
+        {
+            this.fixture = fixture;
+            tableName = fixture.SqlNullValuesTable.Name;
+            // Disable the cache to avoid false failures.
+            SqlConnection.ColumnEncryptionQueryMetadataCacheEnabled = false;
+
+            foreach (string connStr in DataTestUtility.AEConnStringsSetup)
+            {
+                // Insert data and create functions for SqlNullValues test.
+                using (SqlConnection sqlConnection = new SqlConnection(connStr))
+                {
+                    sqlConnection.Open();
+
+                    using (SqlCommand cmd = new SqlCommand(String.Format("INSERT INTO [{0}] (c1) VALUES (@c1)", tableName), sqlConnection, null, SqlCommandColumnEncryptionSetting.Enabled))
+                    {
+                        SqlParameter param = cmd.Parameters.Add("@c1", SqlDbType.Int);
+                        param.Value = DBNull.Value;
+                        cmd.ExecuteNonQuery();
+
+                        param.Value = 10;
+                        cmd.ExecuteNonQuery();
+                    }
+
+                    string sql1 = $"CREATE FUNCTION {UdfName}() RETURNS INT AS \n BEGIN \n RETURN (SELECT c1 FROM [{tableName}] WHERE c1 IS NULL)\n END";
+                    string sql2 = $"CREATE FUNCTION {UdfNameNotNull}() RETURNS INT AS \n BEGIN \n RETURN (SELECT c1 FROM [{tableName}] WHERE c1 IS NOT NULL)\n END";
+                    using (SqlCommand cmd = sqlConnection.CreateCommand())
+                    {
+                        cmd.CommandText = sql1;
+                        cmd.ExecuteNonQuery();
+
+                        cmd.CommandText = sql2;
+                        cmd.ExecuteNonQuery();
+                    }
+                }
+            }
+        }
+
+        [ConditionalTheory(typeof(DataTestUtility), nameof(DataTestUtility.AreConnStringSetupForAE))]
+        [ClassData(typeof(NullValueTestsData))]
+        public void NullValueTests(string connString, ConnectionStringSetting connStringSetting, SqlCommandColumnEncryptionSetting commandSetting, ReturnValueSetting nullReturnValue)
+        {
+            switch (connStringSetting)
+            {
+                case ConnectionStringSetting.Enabled:
+                    connString += "; Column Encryption Setting=Enabled";
+                    break;
+                case ConnectionStringSetting.Disabled:
+                    connString += "; Column Encryption Setting=Disabled";
+                    break;
+            }
+
+            using (SqlConnection sqlConn = new SqlConnection(connString))
+            {
+                sqlConn.Open();
+
+                // Create a command similarly
+                SqlCommand cmd = new SqlCommand(String.Format("SELECT c1 FROM [{0}] ORDER BY c2 ASC", tableName),
```

Comment:
If not using `using`, please add `cmd.close();` and `reader.close()` in `finally` of a try-finally block.

Reply:
'SqlCommand' does not contain a definition for 'Close'. I will use `using` block instead.